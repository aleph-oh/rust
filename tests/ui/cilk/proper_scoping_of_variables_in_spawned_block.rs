#![feature(cilk)]
// Tests that when a variable is declared in a spawned block, it's not usable outside
// that spawned block.
fn main() {
    let _ = cilk_spawn { let y = 5; y };
    println!("y={}", y);
//~^ ERROR cannot find value `y` in this scope [E0425]
}
