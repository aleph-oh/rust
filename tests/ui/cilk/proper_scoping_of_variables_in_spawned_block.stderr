error[E0425]: cannot find value `y` in this scope
  --> $DIR/proper_scoping_of_variables_in_spawned_block.rs:4:22
   |
LL |     println!("y={}", y);
   |                      ^
   |
help: the binding `y` is available in a different scope in the same function
  --> $DIR/proper_scoping_of_variables_in_spawned_block.rs:3:30
   |
LL |     let _ = cilk_spawn { let y = 5; y };
   |                              ^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0425`.
